runtime/caml/fiber.h

```diff
@@ -167,11 +184,6 @@ struct c_stack_link {
  * implemented in the assembly files for an architecture (such as
  * runtime/amd64.S).
  *
- * A continuation object represents a suspended OCaml stack. It contains
- * the stack pointer tagged as an integer to avoid being followed by the GC.
- * In the code the tagged pointer can be referred to as a 'fiber':
- *     fiber := Val_ptr(stack)
- *
  * caml_runstack new_stack function argument
  *  caml_runstack launches a function (with an argument) in a new OCaml
  *  stack. It switches execution from the parent OCaml stack to the fresh
```

Comment:
Looking at this paragraph and below, the existing text clearly uses “stack” and “stack fragment” as synonyms. Do we consider this to be a problem?

This file is called fiber.h but the word fiber is not defined. Should we remedy this?

Finally, in the rest of the repo and notably in `runtime/*.S` files, “fiber” is used to designate heap-allocated blocks consisting of a stack fragment and its metadata.

In light of this, I propose to use:

- the above definition for “fiber”
- for “stack”, the usual concept of call stack in programming
- for “stack fragment”, a fragment of the stack, usually stored either in the system stack or in a fiber.

and modify fiber.h accordingly. Notably, we should try to distinguish the uses of “stack” and “stack fragment” according to these definitions.